### PR TITLE
TF-1246: Enable FCM for Web browser

### PR DESCRIPTION
**Pull request by ManhNTX**
https://user-images.githubusercontent.com/107173849/204537950-c37b427b-17c0-4722-88f0-5a1c5bec34b3.mov

https://user-images.githubusercontent.com/107173849/204726794-8064135f-8d55-47fe-a4bd-b83febf302b4.mov

### CODE REVIEW

File: lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart [hunk 1]
```diff
@@ -358,8 +358,8 @@ class MailboxDashBoardController extends ReloadableController {
       _getUserProfile();
       injectAutoCompleteBindings(sessionCurrent, accountId.value);
       injectRuleFilterBindings(sessionCurrent, accountId.value);
-      injectVacationBindings(sessionCurrent, accountId.value);
       injectFCMBindings(sessionCurrent, accountId.value);
+      injectVacationBindings(sessionCurrent, accountId.value);
```
Comment (hoangdat) [inline lib/features/mailbox_dashboard/presentation/controller/mailbox_dashboard_controller.dart:362]: What is the different?

**Review by hoangdat — COMMENTED** [review]
Please demo the case web app is not in foreground

File: lib/features/push_notification/presentation/services/fcm_receiver.dart [hunk 2]
```diff
@@ -26,7 +29,12 @@ class FcmReceiver {
...
 
   void getFcmToken() async {
-    final token = await FirebaseMessaging.instance.getToken();
-    log('FcmReceiver::onFcmToken():token: $token');
+    try {
+      final token = await FirebaseMessaging.instance.getToken(vapidKey: kIsWeb ? dotenv.get('FIREBASE_WEB_VAPID_PUBLIC_KEY', fallback: '') : null);
+      log('FcmReceiver::onFcmToken():token: $token');
+    } catch(e) {
+      log('FcmReceiver::onFcmToken():exception: $e');
+      throw NotLoadedFCMTokenException();
```
Comment (dab246) [inline lib/features/push_notification/presentation/services/fcm_receiver.dart:37]: You should `try/catch` when calling this function. Avoid crashes

**Review by dab246 — COMMENTED** [review]
Please have a test demo video with the case of `multiple tab`, `close tab`

File: lib/features/push_notification/presentation/services/fcm_receiver.dart (hunk 2 above)
Comment (ManhNTX) [reply lib/features/push_notification/presentation/services/fcm_receiver.dart:37]: it `try/catch`  in `injectFCMBindings`

**Review by dab246 — APPROVED** [review]
lgtm

**Review by hoangdat — APPROVED** [review]

**Pull request merged**